virtualmachine/vsphere/util.go

```diff
@@ -1690,37 +1677,43 @@ func createCustomSpecStaticIp(vm *VM) error {
 }
 
 // updateCustomSpec: updates custom spec structure with the ip settings
-func updateCustomSpec(vm *VM, tempMo *mo.VirtualMachine,
-	customSpec *types.CustomizationSpec) *types.CustomizationSpec {
+func updateCustomSpec(vm *VM,
```

Comment:
It seems that updateCustomSpec is now creating CustomSpect instead of updating. So we should rename it accordingly.